In [2]:
#impoting libraries
import tweepy 
import json

I will regenerate these keys after submitting the task

In [3]:
consumer_key = "Uo7d8jaq9moFWkMtamFm5rXCn"
consumer_secret = "mmBsc8hAO53L7Sfz3Cy52brgSJoyxCs96cOzttTYomK6CXEuQ0"
access_key = "1108422350490329088-uQpTeEnW8AFYgME4v2y39jBIjfJEIb"
access_secret = "Wvwzv92kW6T7uCi5Hp0uFr2JtZ5VrPp5X75CPIMqHp4fj"

Authorization to consumer key and consumer secret

In [4]:
auth = tweepy.OAuthHandler(consumer_key, consumer_secret)

Access to user's access key and access secret

In [5]:
auth.set_access_token(access_key, access_secret)

Calling api

In [6]:
api = tweepy.API(auth)

a list to hold all the Tweets

In [8]:
totaltweets = []

I am using user_timeline method to extract the tweets. we can get only 200 tweets using user_timeline method.
However, we can make successive requests for older tweets. The maximum number of tweets that we can get in a timeline is 3200.

In [9]:
#makings initial requests for most recent 200 tweets
new_tweets = api.user_timeline(screen_name = "midasIIITD",count=200)

In [10]:
totaltweets.extend(new_tweets)

saving the id of the oldest tweet less one

In [11]:
oldest = totaltweets[-1].id - 1

keep extracting tweets until there are no tweets left to extract

In [12]:
while len(new_tweets) > 0:
    new_tweets = api.user_timeline(screen_name = "midasIIITD",count=200,max_id=oldest)
    #save most recent tweets
    totaltweets.extend(new_tweets)
    #update the id of the oldest tweet less one
    oldest = totaltweets[-1].id - 1

In [13]:
print(len(totaltweets))

296


Total 296 tweets get extracted

Now dumping these tweets to a json file after collecting it to dictionary.

In [16]:
temp={} 
for i,tweet in enumerate(totaltweets):
    temp[i]=tweet._json

In [17]:
with open('total_tweets.json', 'w') as fp:
    json.dump(temp, fp)